src/modules/75-cf/pages/feature-flags/components/AllEnvironmentsFlagsListing.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useMemo } from 'react'
+import type { Cell, Column } from 'react-table'
+import { Container, Layout, TableV2, Text, Utils } from '@harness/uicore'
+import { Color, FontVariation } from '@harness/design-system'
+import type { MutateRequestOptions } from 'restful-react/dist/Mutate'
+import type { AllEnvironmentsFlags, AllEnvironmentsFlag, DeleteFeatureFlagQueryParams, Feature } from 'services/cf'
+import type { EnvironmentResponseDTO } from 'services/cd-ng'
+import { useStrings } from 'framework/strings'
+import FlagOptionsMenuButton from '@cf/components/FlagOptionsMenuButton/FlagOptionsMenuButton'
+import { useFFGitSyncContext } from '@cf/contexts/ff-git-sync-context/FFGitSyncContext'
+import { formatDate } from '@cf/utils/CFUtils'
+import FlagEnvironmentsState from './FlagEnvironmentsState'
+
+export interface AllEnvironmentsFlagsListingProps {
+  environments: EnvironmentResponseDTO[]
+  allEnvironmentsFlags?: AllEnvironmentsFlags | null
+  refetchFlags: () => void
+  deleteFlag: (
+    data: string,
+    mutateRequestOptions?: MutateRequestOptions<DeleteFeatureFlagQueryParams, void> | undefined
```

Comment:
Marking as optional should be enough to not need the `undefined`
```suggestion
    mutateRequestOptions?: MutateRequestOptions<DeleteFeatureFlagQueryParams, void>
```